static/app/components/globalDrawer/index.spec.tsx

```diff
@@ -0,0 +1,198 @@
+      <button data-test-id="drawer-test-close" onClick={closeDrawer}>
+        Close
+      </button>
+    </div>
+  );
+}
+
+describe('GlobalDrawer', function () {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('useDrawer hook can open and close the Drawer', async function () {
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({Body}) => (
+            <Body data-test-id="drawer-test-content">useDrawer hook</Body>
+          ),
+        }}
+      />
+    );
+
+    expect(
+      screen.queryByRole('complementary', {name: 'slide out drawer'})
+    ).not.toBeInTheDocument();
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    expect(await screen.findByTestId('drawer-test-content')).toBeInTheDocument();
+    expect(
+      screen.getByRole('complementary', {name: 'slide out drawer'})
+    ).toBeInTheDocument();
+
+    await userEvent.click(screen.getByTestId('drawer-test-close'));
+
+    expect(screen.queryByTestId('drawer-test-content')).not.toBeInTheDocument();
+    expect(screen.queryByRole('complementary')).not.toBeInTheDocument();
+    expect(
+      screen.queryByRole('complementary', {name: 'slide out drawer'})
+    ).not.toBeInTheDocument();
+  });
+
+  it('calls onClose handler when close button is clicked', async function () {
+    const closeSpy = jest.fn();
+
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({Body}) => (
+            <Body data-test-id="drawer-test-content">onClose button</Body>
+          ),
+          options: {onClose: closeSpy},
+        }}
+      />
+    );
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    expect(await screen.findByTestId('drawer-test-content')).toBeInTheDocument();
+
+    await userEvent.click(screen.getByRole('button', {name: 'Close Drawer'}));
+
+    expect(closeSpy).toHaveBeenCalled();
+    expect(screen.queryByTestId('drawer-test-content')).not.toBeInTheDocument();
+  });
+
+  it('calls onClose handler when clicking outside the drawer', async function () {
+    const closeSpy = jest.fn();
+
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({Body}) => (
+            <Body data-test-id="drawer-test-content">onClose outside click</Body>
+          ),
+          options: {onClose: closeSpy},
+        }}
+      />
+    );
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    expect(await screen.findByTestId('drawer-test-content')).toBeInTheDocument();
+
+    await userEvent.click(screen.getByTestId('drawer-test-outside'));
+
+    expect(closeSpy).toHaveBeenCalled();
+    expect(screen.queryByTestId('drawer-test-content')).not.toBeInTheDocument();
+  });
+
+  // TODO(Leander): I'll have to look into this further in a separate PR.
+  // The useHotKeys hook doesn't seem to register the userEvents for this test
+  it.skip('calls onClose handler when escape key is pressed', async function () {
```

Comment:
@evanpurkhiser Do you know why this may not be working? Manually setting the event listeners had this test passing, but with the `useHotKeys` hook it doesn't seem to be registering the `userEvent.keyboard` signal.

It works in the browser but I couldn't figure out a way to get this test passing again.

Reply:
hrm. I'm not sure. Might just have to investigate what `useHotKeys` is doing